src/shared/lockfile-util.c

```diff
@@ -23,13 +23,7 @@ int make_lock_file(const char *p, int operation, LockFile *ret) {
         assert(p);
         assert(ret);
 
-        /*
-         * We use UNPOSIX locks if they are available. They have nice semantics, and are mostly compatible
-         * with NFS. However, they are only available on new kernels. When we detect we are running on an
-         * older kernel, then we fall back to good old BSD locks. They also have nice semantics, but are
-         * slightly problematic on NFS, where they are upgraded to POSIX locks, even though locally they are
-         * orthogonal to POSIX locks.
-         */
+        /* We use UNPOSIX locks as they have nice semantics, and are mostly compatible with NFS. */
```

Comment:
hmm, UNPOSIX locks work on all file systems? are you sure?

Reply:
I checked the kernel source code and none of the posix/unposix lock logic seems filesystem specific to me. I can't find anything on the internet about some filesystems not supporting this either.

Reply:
ok